File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        

        res=[]

        def backtrac(idx,arr,tot):
            if tot==target:
                res.append(arr[:])
                return
            if tot>target or len(nums)<=idx:
                return
            
            arr.append(nums[idx])
            backtrac(idx,arr,tot+nums[idx])
            arr.pop()
            backtrac(idx+1,arr,tot)


        

        backtrac(0,[],0)


        return res
        

File: Data Structures & Algorithms/reverse-bits/submission-0.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        int res=0;
        vector<int> arr(32,0);

        for(int i=31;i>=0 && n ;i--){
            if(n%2==1){
                arr[i]=1;
            }
            else{
                arr[1];
            }
            res=res+arr[i]*pow(2,i);
            n/=2;
        }

       return res;
        
    }
};
